js/customize-snapshots.js

```diff
@@ -1,104 +1,121 @@
-/* global jQuery, wp, JSON, _customizeSnapshotsSettings */
-/* eslint no-magic-numbers: [ "error", { "ignore": [0,1,-1] } ], consistent-this: [ "error", "snapshot" ] */
+/* global wp, $ */
+/* eslint consistent-this: [ "error", "snapshot", "control" ] */
 
-(function( api, $ ) {
+(function( api ) {
 	'use strict';
 
-	var escKeyCode = 27;
-
 	api.Snapshots = api.Class.extend( {
 
 		data: {
-			action: '',
-			uuid: '',
-			editLink: '',
+			inspectLink: '',
 			title: '',
-			publishDate: '',
-			postStatus: '',
-			currentUserCanPublish: true,
-			initialServerDate: '',
-			initialServerTimestamp: 0,
-			initialClientTimestamp: 0,
-			previewingTheme: '',
-			i18n: {},
-			dirty: false
+			i18n: {
+				title: '',
+				savePending: '',
+				pendingSaved: ''
+			}
 		},
 
-		uuidParam: 'customize_changeset_uuid',
-
 		initialize: function initialize( snapshotsConfig ) {
 			var snapshot = this;
 
-			snapshot.schedule = {};
-
 			if ( _.isObject( snapshotsConfig ) ) {
 				_.extend( snapshot.data, snapshotsConfig );
 			}
 
-			// Set the initial client timestamp.
-			snapshot.data.initialClientTimestamp = snapshot.dateValueOf();
-
 			api.bind( 'ready', function() {
-				api.state.create( 'snapshot-exists', false );
-				api.state.create( 'snapshot-saved', true );
-				api.state.create( 'snapshot-submitted', true );
-
-				snapshot.data.uuid = snapshot.data.uuid || api.settings.changeset.uuid;
-				snapshot.data.title = snapshot.data.title || snapshot.data.uuid;
+				var saveBtn = $( '#save' );
 
-				snapshot.editBoxAutoSaveTriggered = false;
+				snapshot.data.title = snapshot.data.title || api.settings.changeset.uuid;
+				api.state.create( 'changesetTitle', snapshot.data.title );
+				api.state.create( 'changesetInspectLink', snapshot.data.inspectLink );
 
-				if ( api.state.has( 'changesetStatus' ) && api.state( 'changesetStatus' ).get() ) {
-					api.state( 'snapshot-exists' ).set( true );
-				}
+				api.control( 'changeset_scheduled_date', snapshot.setupScheduledChangesetCountdown );
 
-				snapshot.extendPreviewerQuery();
+				api.bind( 'save-request-params', function( data ) {
+					if ( api.state( 'selectedChangesetStatus' ) && 'publish' !== api.state( 'selectedChangesetStatus' ) ) {
```

Comment:
The `publish` check doesn't seem right here. The title should be sent regardless of the status.